mbq/metrics/contrib/django/middleware/timing.py

```diff
@@ -0,0 +1,50 @@
+from time import time
+
+from django.conf import settings
+try:
+    from django.utils.deprecation import MiddlewareMixin as MiddlewareDeprecationMixin
+except ImportError:
+    MiddlewareDeprecationMixin = object
+
+import mbq.metrics as metrics
+
+DEFAULT_SETTINGS = {
+    'EXCLUDED_PATHS': []
+}
+
+SETTINGS = DEFAULT_SETTINGS.copy()
+SETTINGS.update(
+    getattr(settings, 'MBQ_METRICS', {})
+)
+
+
+class TimingMiddleware(MiddlewareDeprecationMixin):
+
+    def __init__(self, get_response=None):
+        self.get_response = get_response
+
+    def process_request(self, request):
+        if request.path in SETTINGS['EXCLUDED_PATHS']:
+            return request
+        setattr(request, 'mbq-metrics-start-time', time())
```

Comment:
Let's throw an underscore at the front to make sure no one things it's something for general use.

Reply:
✅